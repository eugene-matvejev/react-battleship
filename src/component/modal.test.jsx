import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _Modal from './modal';

const { WrappedComponent: Modal } = _Modal;

configure({ adapter: new Adapter() });

describe('<Modal/>', () => {
    const props = {
        history: {
            goBack: jest.fn(),
        },
        title: '{{label}}',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Modal {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
                ['children', '{{children}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Modal {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });

    describe('external callbacks', () => {
        describe('::history.goBack', () => {
            it('should be invoked from a click on a [data-cy="modal-close"]', () => {
                shallow(<Modal {...props} />).find('[data-cy="modal-close"]').simulate('click');

                expect(props.history.goBack).toBeCalled();
            });
        });
    });
});
