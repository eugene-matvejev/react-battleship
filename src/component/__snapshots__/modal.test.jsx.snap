// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> render with default/required props 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal "
  >
    <button
      className="modal_button--close"
      data-cy="modal-close"
      onClick={[MockFunction]}
    />
    <h1
      className="modal_title"
      data-cy="modal-title"
    >
      {{label}}
    </h1>
  </section>
</div>
`;

exports[`<Modal/> render with optional props [::children] as "{{children}}" 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal "
  >
    <button
      className="modal_button--close"
      data-cy="modal-close"
      onClick={[MockFunction]}
    />
    <h1
      className="modal_title"
      data-cy="modal-title"
    >
      {{label}}
    </h1>
    {{children}}
  </section>
</div>
`;

exports[`<Modal/> render with optional props [::className] as "{{className}}" 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal {{className}}"
  >
    <button
      className="modal_button--close"
      data-cy="modal-close"
      onClick={[MockFunction]}
    />
    <h1
      className="modal_title"
      data-cy="modal-title"
    >
      {{label}}
    </h1>
  </section>
</div>
`;

exports[`<Modal/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal "
  >
    <button
      className="modal_button--close"
      data-cy="{{data-cy}}modal-close"
      onClick={[MockFunction]}
    />
    <h1
      className="modal_title"
      data-cy="{{data-cy}}modal-title"
    >
      {{label}}
    </h1>
  </section>
</div>
`;
